### PR TITLE
2 BE and 3 FE tests added, and Pages are re-organized with PEP8 guidelines

### PR DESCRIPTION
**Pull request by Rohanksingh18**
Kindly review.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  EDGE_REMOTE_WEBDRIVER_URL: http://edge_remote_browser:4444/wd/hub
+  OPERA_REMOTE_WEBDRIVER_URL: http://opera_remote_browser:4444/wd/hub
+
+
+run_ecom_backend:
+  stage:
+    run_tests
+  image: python:3.11
+  script:
+    - echo "STRING BACKEND TESTS....."
```
Comment (admaskinfu) [inline .gitlab-ci.yml]: You mean 'Starting'? typo

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    - pip3 install -r ./requirements.txt
+    - source ./variables.sh
+    - /bin/sh ./runner.sh ecom_test/tests/backend_tests "-m smoke"
+    - echo "DONE RUNNING BACKEND TESTS....."
+  artifacts:
+    when: always
+    paths:
+      - ./output
+
+run_ecom_smoke_frontend:
```
Comment (admaskinfu) [inline .gitlab-ci.yml:32]: come up with a better name.
May be have 2 jobs (run_ecom_smoke_frontend, run_ecom_regression_frontend, run_ecom_smoke_backend, run_ecom_regression_backend)

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    run_tests
+  image: python:3.11
+  variables:
+    REMOTE_WEBDRIVER: $CHROME_REMOTE_WEBDRIVER_URL
+    BROWSER: remote_chrome
+  services:
+    - name: selenium/standalone-chrome:4.7.1-20221208
+      alias: chrome_remote_browser
+  script:
+    - echo "STRING FRONTEND SMOKE TESTS....."
```
Comment (admaskinfu) [inline .gitlab-ci.yml]: typo

File: README.md
Comment (admaskinfu) [inline README.md:1]: Why is this deleted?

File: ecom_test/src/api_helpers/CouponsAPIHelper.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        return self.woo_api_utility.post('coupons', params=payload, expected_status_code=201)
+
+    #  Calling the get method of the woo_api_utility instance with the coupon ID as a parameter.
+    def call_retrieve_coupon(self, coupon_id):
+        logger.debug(f"Calling retrieve a coupon. Coupon id: {coupon_id}")
+        return self.woo_api_utility.get(f'coupons/{coupon_id}')
+
+    # calling the get method of the woo_api_utility instance.
+    def call_list_all_coupons(self, payload=None):
+        logger.debug("Calling list all coupons.")
```
Comment (admaskinfu) [inline ecom_test/src/api_helpers/CouponsAPIHelper.py:23]: the messaging should be more readable like "Calling 'list all coupons'." Just adding the inside quotes.

File: ecom_test/src/api_helpers/CouponsAPIHelper.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+from ecom_test.src.utilities.wooAPIUtility import WooAPIUtility
+import logging as logger
+
+
+class CouponsAPIHelper:
+    # class has an __init__ method that initializes an instance of the WooAPIUtility class.
+    def __init__(self):
+        self.woo_api_utility = WooAPIUtility()
+
+    # Method makes a POST request to create a new coupon by calling the post-method of the woo_api_utility instance.
```
Comment (admaskinfu) [inline ecom_test/src/api_helpers/CouponsAPIHelper.py]: Use proper docstring and not comments. You MUST use doc strings. Under the function name use 3 quotes to add doc string. do a quick look up of doc strings.
Example
`def call_create_coupon(self, payload):
       """
       Makes a POST request to create a new coupon by calling the post-method of the woo_api_utility instance.
       """`

File: ecom_test/src/api_helpers/ProductsAPIHelpers.py [hunk 6]
```diff
@@ -2,8 +2,8 @@
 
 from ecom_test.src.utilities.wooAPIUtility import WooAPIUtility
 
-#Create a class for product API
 
+# Create a class for product API
```
Comment (admaskinfu) [inline ecom_test/src/api_helpers/ProductsAPIHelpers.py]: Add it as doc string not comment.

File: ecom_test/src/pages/HomePage.py [hunk 7]
```diff
@@ -19,5 +17,11 @@ def go_to_the_home_page(self):
     def click_add_to_cart_btn(self):
         self.sl.wait_and_click(self.Add_To_Cart_BTN)
 
+    def get_all_product_elements(self):
+        return self.sl.wait_and_get_elements(self.Product_Elements)
```
Comment (admaskinfu) [inline ecom_test/src/pages/HomePage.py]: For the locator variables, use upper case letter. It is a convention to use upper case for constants. Those are variables that will not change so they are constants. It looks much better.
If you do not agree with that then make all lower case. Definitely do not do 'Product_Elements'.

File: ecom_test/src/pages/locators/MyAccountSignedInPageLocators.py [hunk 8]
```diff
@@ -1,5 +1,6 @@
 
 from selenium.webdriver.common.by import By
 
+
 class MyAccountSignedInPageLocator:
     Left_Nav_LogOut_BTN = (By.XPATH, "//a[normalize-space()='Logout']") #to verfiy logOut button is visbile after registration
```
Comment (admaskinfu) [inline ecom_test/src/pages/locators/MyAccountSignedInPageLocators.py]: Use upper case for the constants (these variables are constant)

File: ecom_test/tests/backend_tests/coupons/test_create_coupons.py [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+from ecom_test.src.utilities.wooAPIUtility import WooAPIUtility
+
+
+# These markers can be used to selectively run or exclude tests based on their assigned markers.
+pytestmark = [pytest.mark.regression, pytest.mark.coupons]
+
+
+@pytest.fixture(scope='module')
+def my_setup():
+    """
```
Comment (admaskinfu) [inline ecom_test/tests/backend_tests/coupons/test_create_coupons.py:14]: This is a good example of using doc string.

File: variables.sh [hunk 10]
```diff
@@ -1,11 +1,9 @@
...
-set BROWSER=chrome
-set RESULTS_DIR=$(pwd)/results
-set DB_PORT_OVERRIDE=8889
-set DB_HOST_OVERRIDE=localhost
-set DB_DATABASE_OVERRIDE=ecomsite
-set DB_TABLE_PREFIX_OVERRIDE=wp_
-set WOO_KEY=ck_b7504831538d902f98ebdb3c97e541412b1c3970
-set WOO_SECRET=cs_778cd37bb50c506ba42f964b9094a37cc21af3a3
\ No newline at end of file
+export BASE_URL=http://localhost:8888/EcomSite/
```
Comment (admaskinfu) [inline variables.sh:3]: You can keep 2 variable files. One for your local (Windows version) and one for Linux version.

File: .gitlab-ci.yml (hunk 2 above)
Comment (Rohanksingh18) [reply .gitlab-ci.yml:32]: Changed.

File: README.md
Comment (Rohanksingh18) [reply README.md:1]: Created here by mistake

File: ecom_test/src/api_helpers/CouponsAPIHelper.py (hunk 4 above)
Comment (Rohanksingh18) [reply ecom_test/src/api_helpers/CouponsAPIHelper.py:23]: changed

File: ecom_test/src/api_helpers/CouponsAPIHelper.py (hunk 5 above)
Comment (Rohanksingh18) [reply ecom_test/src/api_helpers/CouponsAPIHelper.py]: changed

File: ecom_test/tests/backend_tests/coupons/test_create_coupons.py (hunk 9 above)
Comment (Rohanksingh18) [reply ecom_test/tests/backend_tests/coupons/test_create_coupons.py:14]: Thanks

File: variables.sh (hunk 10 above)
Comment (Rohanksingh18) [reply variables.sh:3]: Yes, I have but it's only in the local only as I added it into gitignore. However, I will push that well.

File: ecom_test/src/pages/locators/MyAccountSignedInPageLocators.py (hunk 8 above)
Comment (Rohanksingh18) [reply ecom_test/src/pages/locators/MyAccountSignedInPageLocators.py]: changed

File: ecom_test/src/pages/HomePage.py (hunk 7 above)
Comment (Rohanksingh18) [reply ecom_test/src/pages/HomePage.py]: made change for every locators

File: ecom_test/src/api_helpers/ProductsAPIHelpers.py (hunk 6 above)
Comment (Rohanksingh18) [reply ecom_test/src/api_helpers/ProductsAPIHelpers.py]: done

File: .gitlab-ci.yml (hunk 3 above)
Comment (Rohanksingh18) [reply .gitlab-ci.yml]: rectified

File: .gitlab-ci.yml (hunk 2 above)
Comment (Rohanksingh18) [reply .gitlab-ci.yml:32]: suggestions applied

**Review by admaskinfu — APPROVED** [review]

File: README.md
Comment (Rohanksingh18) [inline README.md:1]: done

File: desktop.ini
Comment (Rohanksingh18) [inline desktop.ini:1]: done

**Comment by Rohanksingh18** [comment]
All done
